src/dynsight/_internal/trajectory/trajectory.py

```diff
@@ -0,0 +1,461 @@
+                file_path,
+                data_insight.dataset,
+                self.state_list,
+                self.labels,
+                self.meta["delta_t"],
+            )
+
+    def plot_one_trj(
+        self,
+        file_path: Path,
+        data_insight: Insight,
+        particle_id: int,
+    ) -> None:
+        """Plot one particle's trajectory colored according to clustering."""
+        if data_insight.dataset.ndim == UNIVAR_DIM:
+            dynsight.onion.plot.plot_one_trj_uni(
+                file_path,
+                particle_id,
+                self.reshaped_data,
+                data_insight.dataset.shape[0],
+                self.labels,
+            )
+        else:
+            dynsight.onion.plot.plot_one_trj_multi(
+                file_path,
+                particle_id,
+                self.meta["delta_t"],
+                data_insight.dataset,
+                self.labels,
+            )
+
+    def plot_medoids(self, file_path: Path, data_insight: Insight) -> None:
+        """Plot the average sequence of each onion cluster."""
+        if data_insight.dataset.ndim == UNIVAR_DIM:
+            dynsight.onion.plot.plot_medoids_uni(
+                file_path,
+                self.reshaped_data,
+                self.labels,
+            )
+        else:
+            dynsight.onion.plot.plot_medoids_multi(
+                file_path,
+                self.meta["delta_t"],
+                data_insight.dataset,
+                self.labels,
+            )
+
+    def plot_state_populations(
+        self,
+        file_path: Path,
+        data_insight: Insight,
+    ) -> None:
+        """Plot each state's population along the trajectory."""
+        dynsight.onion.plot.plot_state_populations(
+            file_path,
+            data_insight.dataset.shape[0],
+            self.meta["delta_t"],
+            self.labels,
+        )
+
+    def plot_sankey(
+        self,
+        file_path: Path,
+        data_insight: Insight,
+        frame_list: list[int],
+    ) -> None:
+        """Plot the Sankey diagram of the onion clustering."""
+        dynsight.onion.plot.plot_sankey(
+            file_path,
+            self.labels,
+            data_insight.dataset.shape[0],
+            frame_list,
+        )
+
+
+@dataclass(frozen=True)
+class Trj:
+    """Contains a trajectory.
+
+    Attributes:
+        universe: a MDAnalysis.Universe containing the trajectory.
+
+    .. warning::
+
+        This class is under development. The name and type of the "universe"
+        attribute may change in the future.
+    """
+
+    universe: MDAnalysis.Universe = field()
+
+    @classmethod
+    def init_from_universe(cls, universe: MDAnalysis.Universe) -> Trj:
+        """Initialize Trj object from MDAnalysis.Universe."""
+        return Trj(universe)
+
+    @classmethod
+    def init_from_xyz(cls, traj_file: Path, dt: float) -> Trj:
+        """Initialize Trj object from .xyz file."""
```

Comment:
Actually, in doc strings of functions that go straight to mdanalysis, can you link to the mdanalysis function it uses that explains its parameters?